src/App.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import './css/App.css';
+import Header from './Header';
+import Timer from './Timer';
+import Image from './Image';
+import {TimerState, TimerType, TimerConstants, MAX_SESSION_TIME, MIN_SESSION_TIME, MAX_BREAK_TIME, MIN_BREAK_TIME} from './Constants';
+
+class App extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            breakLength:TimerConstants.breakLength,
+            sessionLength:TimerConstants.sessionLength,
+            type: TimerType.SESSION,
+            timerState: TimerState.NOTSTARTED
+        }
+        this.increaseSessionLength = this.increaseSessionLength.bind(this);
+        this.decreaseSessionLength = this.decreaseSessionLength.bind(this);
+        this.increaseBreakLength = this.increaseBreakLength.bind(this);
+        this.decreaseBreakLength = this.decreaseBreakLength.bind(this);
+    }
+
+    reset = () => {
+        this.setState({
+            breakLength:TimerConstants.breakLength,
+            sessionLength:TimerConstants.sessionLength,
+            type: TimerType.SESSION,
+            timerState: TimerState.RESET
+        });
+    }
+
+    increaseSessionLength = () => {
+        if(this.state.timerState !== TimerState.START) {
+            this.setState({sessionLength: (this.state.sessionLength < MAX_SESSION_TIME )?this.state.sessionLength+1:this.state.sessionLength});
+        }
+    }
+
+    decreaseSessionLength = () => {
+        if(this.state.timerState !== TimerState.START) {
+            this.setState({sessionLength: (this.state.sessionLength > MIN_SESSION_TIME)?this.state.sessionLength-1:this.state.sessionLength});
+        }
+    }
+
+    increaseBreakLength = () => {
+        if(this.state.timerState !== TimerState.START) {
+            this.setState({breakLength: (this.state.breakLength < MAX_BREAK_TIME)?this.state.breakLength+1:this.state.breakLength});
+        }
+    }
+
+    decreaseBreakLength = () => {
+        if(this.state.timerState !== TimerState.START) {
+            this.setState({breakLength: (this.state.breakLength > MIN_BREAK_TIME)?this.state.breakLength-1:this.state.breakLength });
+        }
+    }
+
+    setTimerState = (timerState) => {
+        this.setState({timerState: timerState});
```

Comment:
why is sessionTime and sessionType is being set by Timer. I had hoped Timer's only work is to display current time and session type and not setting these.